File: core/runtime.go/prelude.go
/* This is free and unencumbered software released into the public domain. */

// Package dry provides DRYlib for Go.
package dry

////////////////////////////////////////////////////////////////////////////////
// Type Definitions

// Boolean (true or false)
type Bool = bool

// Character
type Char = rune

// Complex number (arbitrary size)
type Complex struct {
	real      Real
	imaginary Real
}

// Floating-point number (native size)
type Float = float64

// Floating-point number (32-bit single-precision)
type Float32 = float32

// Floating-point number (64-bit double-precision)
type Float64 = float64

// Integer number (native size)
type Int = int

// Integer number (8-bit)
type Int8 = int8

// Integer number (16-bit)
type Int16 = int16

// Integer number (32-bit)
type Int32 = int32

// Integer number (64-bit)
type Int64 = int64

// Integer number (128-bit)
type Int128 = Integer

// Integer number (arbitrary size)
type Integer struct {
	value int64 // FIXME
}

// Natural number (arbitrary size)
type Natural = Integer

// Rational number (arbitrary size)
type Rational struct {
	numerator   Integer
	denominator Integer
}

// Real number (arbitrary size)
type Real = struct {
	value Float64 // FIXME
}

// Machine word (native size)
type Word = uint

// Machine word (8-bit)
type Word8 = uint8

// Machine word (16-bit)
type Word16 = uint16

// Machine word (32-bit)
type Word32 = uint32

// Machine word (64-bit)
type Word64 = uint64

File: core/runtime.jl/prelude.jl
# This is free and unencumbered software released into the public domain.

################################################################################
# Type Definitions

# Boolean (true or false)
const Bool = Core.Bool

# Character (Unicode code point)
const Char = Core.Char

# Complex number (arbitrary size)
const Complex = Base.Complex{Base.MPFR.BigFloat}

# Floating-point number (native size)
const Float = Core.Float64

# Floating-point number (32-bit single-precision)
const Float32 = Core.Float32

# Floating-point number (64-bit double-precision)
const Float64 = Core.Float64

# Integer number (native size)
const Int = if (Core.Int === Core.Int64) Core.Int64 else Core.Int32 end

# Integer number (8-bit)
const Int8 = Core.Int8

# Integer number (16-bit)
const Int16 = Core.Int16

# Integer number (32-bit)
const Int32 = Core.Int32

# Integer number (64-bit)
const Int64 = Core.Int64

# Integer number (128-bit)
const Int128 = Core.Int128

# Integer number (arbitrary size)
const Integer = Base.GMP.BigInt

# Natural number (arbitrary size)
const Natural = Integer

# Rational number (arbitrary size)
const Rational = Base.Rational{Base.GMP.BigInt}

# Real number (arbitrary size)
const Real = Base.MPFR.BigFloat

# Machine word (native size)
const Word = if (Core.UInt === Core.UInt64) Core.UInt64 else Core.UInt32 end

# Machine word (8-bit)
const Word8 = Core.UInt8

# Machine word (16-bit)
const Word16 = Core.UInt16

# Machine word (32-bit)
const Word32 = Core.UInt32

# Machine word (64-bit)
const Word64 = Core.UInt64

# Machine word (128-bit)
const Word128 = Core.UInt128

#println("OK") # julia prelude.jl
